diambra/arena/arena_gym.py

```diff
@@ -2,95 +2,137 @@
-        self.observation_space = spaces.Box(low=0, high=255,
-                                            shape=(self.hwc_dim[0],
-                                                   self.hwc_dim[1],
-                                                   self.hwc_dim[2]),
-                                            dtype=np.uint8)
-
-    # Processing Environment info
-    def env_info_process(self, env_info_dict):
-        # N actions
-        self.n_actions_but_comb = env_info_dict["n_actions"][0]
-        self.n_actions_no_but_comb = env_info_dict["n_actions"][1]
         # N actions
-        if self.env_settings.player == "P1P2":
-            self.n_actions = [self.n_actions_but_comb, self.n_actions_but_comb]
-            for idx in range(2):
-                if self.attack_but_combination[idx] is False:
-                    self.n_actions[idx] = self.n_actions_no_but_comb
-        else:
-            self.n_actions = self.n_actions_but_comb
-            if self.attack_but_combination is False:
-                self.n_actions = self.n_actions_no_but_comb
+        self.n_actions = [self.env_info.available_actions.n_moves, self.env_info.available_actions.n_attacks]
+
+        # Actions tuples and dict
+        move_tuple = ()
+        move_dict = {}
+        attack_tuple= ()
+        attack_dict = {}
 
-        # Frame height, width and channel dimensions
-        self.hwc_dim = env_info_dict["frame_shape"]
+        for idx in range(len(self.env_info.available_actions.moves)):
+            move_tuple += (self.env_info.available_actions.moves[idx].key,)
+            move_dict[idx] = self.env_info.available_actions.moves[idx].label
+
+        for idx in range(len(self.env_info.available_actions.attacks)):
+            attack_tuple += (self.env_info.available_actions.attacks[idx].key,)
+            attack_dict[idx] = self.env_info.available_actions.attacks[idx].label
+
+        self.actions_tuples = (move_tuple, attack_tuple)
+        self.print_actions_dict = [move_dict, attack_dict]
 
         # Maximum difference in players health
-        self.max_delta_health = env_info_dict["delta_health"]
+        for k in sorted(self.env_info.ram_states.keys()):
+            if "health" in k:
+                self.max_delta_health = self.env_info.ram_states[k].max - self.env_info.ram_states[k].min
+                break
+
+    def _get_ram_states_obs_dict(self):
+        player_spec_dict = {}
+        generic_dict = {}
+        # Adding env additional observations (side-specific)
+        for k, v in self.env_info.ram_states.items():
+            if k[-2:] == "P1":
+                target_dict = player_spec_dict
+                knew = "own_" + k[:-2]
+            elif k[-2:] == "P2":
+                target_dict = player_spec_dict
+                knew = "opp_" + k[:-2]
+            else:
+                target_dict = generic_dict
+                knew = k
+
+            # Discrete spaces (binary / categorical)
+            if v.type == 0 or v.type == 2:
```

Comment:
Using a const for these magic number would be better. Ideally coming from the protobuf side if possible

Reply:
I see your point and this also bugs me. I will try to think about how to better handle them and if I will find a good solution quickly. Will keep you posted.

Reply:
Done, looks way better now: take a look here and let me know: https://github.com/diambra/arena/commit/0770bff1e963cc9baecf06833a4555388c35557c

Reply:
Perfect!